core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/udf/python/source/PythonUDFSourceOpDescV2.java

```diff
@@ -82,10 +86,15 @@ public PhysicalOp getPhysicalOp(WorkflowIdentity workflowId, ExecutionIdentity e
                 .withLocationPreference(Option.empty());
 
 
-        if (workers > 1) {
-            return physicalOp
-                    .withParallelizable(true)
-                    .withSuggestedWorkerNum(workers);
+        if (parallelizable) {
+            if (workers != null) {
+                return physicalOp
+                        .withParallelizable(true)
+                        .withSuggestedWorkerNum(workers);
+            } else {
+                return physicalOp
+                        .withParallelizable(true);
+            }
```

Comment:
```
physicalOp = physicalOp.withParallelizable(parallelizable);

if (workers!=null) {
    physicalOp = physicalOp.withSuggestedWorkerNum(workers);
}

return physicalOp;
```

Reply:
add an assertion to prevent from parallelizable == true and worker != null